src/devices_models/devices/common/cost_functions.jl

```diff
@@ -49,6 +49,17 @@ function AddCostSpec(
     error("AddCostSpec is not implemented for $T / $U")
 end
 
+function add_service_variables!(
+    spec::AddCostSpec,
+    services::Vector{T},
+) where {T <: PSY.Service}
+    for service in services
+        name = PSY.get_name(service)
+        spec.addtional_linear_terms[name] = make_variable_name(name, typeof(service))
```

Comment:
We should do this with a setter function just in case the implementation changes later.